src/app/helpers/filters/audioRecordingFilters.ts

```diff
@@ -0,0 +1,113 @@
+import { DateExpressions, InnerFilter } from "@baw-api/baw-api.service";
+import { AudioRecording } from "@models/AudioRecording";
+import { DateTime } from "luxon";
+import { filterAnd } from "./filters";
+
+/**
+ * Adds date range conditions to an existing filter
+ *
+ * @param startDate (optional) The minimum `recordedDate` of the date range
+ * @param endDate (optional) The maximum `recordedEndDate` allowed
+ * @param filters (optional) An existing filter to add the date range conditions to. If no filter is provided, a new filter will be created
+ * @returns A new filter with all the same conditions as the initial filter, with the date range conditions added
+ */
+export function filterDate(
+  startDate?: DateTime,
+  endDate?: DateTime,
+  filters?: InnerFilter
+): InnerFilter {
+  if (startDate) {
+    // the baw-api is expecting the date in the ISO8601 format.
+    const startDateFilter: InnerFilter<AudioRecording> = {
+      recordedDate: { greaterThanOrEqual: startDate.toJSON() },
+    };
+
+    filters = filterAnd(filters, startDateFilter);
+  }
+
+  if (endDate) {
+    const endDateFilters = {
+      recordedEndDate: { lessThan: endDate.toJSON() },
+    };
+
+    filters = filterAnd(filters, endDateFilters);
+  }
+
+  return filters;
+}
+
+/**
+ * Adds time range conditions to an existing filter
+ *
+ * @param expressions Describes if the time filters use local (daylight savings) time
+ * @param startTime (optional) The minimum `recordedDate` allowed
+ * @param endTime (optional) The maximum `recordedEndDate` allowed
+ * @param filters (optional) An existing filter to add the time range conditions to. If no filter is provided, a new filter will be created
+ * @returns
+ */
+export function filterTime(
+  ignoreDayLightSavings: boolean,
+  startTime?: string,
+  endTime?: string,
+  filters?: InnerFilter
+): InnerFilter {
+  const expressions: DateExpressions[] = ignoreDayLightSavings
+    ? ["local_offset", "time_of_day"]
+    : ["local_tz", "time_of_day"];
+
+  // a midnight filter is a filter circumstance where the time range overlaps two days e.g. 10PM to 1AM
+  // midnight filters can only occur if both a start and end time are specified
+  // therefore, this statement should return false if both start and end times are not specified
+  const isMidnightFilter = startTime && endTime && startTime > endTime;
```

Comment:
this is incorrect - you're doing string comparisons on numbers. Thinking about this, it might hold actually (because the strings are always the same length) but regardless this is really bad practice.